File: Python/151__Reverse_Words_in_a_String.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        s += " " #in case, no trailing spaces
        words = []
        word = ""
        for i in range(len(s)):
            if s[i] != " ":
                word += s[i]
            else:
                if word:
                    words.append(word)
                word = ""
        result = ""
        while words:
            result += words.pop()
            if words:
                result += " "
        return result

# Simpler Solution
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        return " ".join(s.split()[::-1])

File: Java/151__Reverse_Words_in_a_String.java
class Solution {
    public String reverseWords(String s) {
        StringBuilder sb = new StringBuilder();
        String[] strings = s.trim().split(" ");
        for (int i = strings.length - 1; i >= 0; i--) {
            if (strings[i].length() > 0) {
                sb.append(strings[i]);
                if (i > 0) sb.append(" ");
            }
        }
        return sb.toString();
    }
}
